File: server/apps/subjects/application/dto/subject_dto.py
from dataclasses import dataclass
from decimal import Decimal


@dataclass
class SubjectDTO:
    subject_id: str
    name: str

    university_id: str
    university_name: str

    department_id: str
    department_name: str

    subject_types: str
    credits: int

    process_percent: Decimal
    midterm_percent: Decimal
    final_percent: Decimal

    status: str

File: client/src/application/subjects/dto/subject.dto.js
class SubjectDTO {
  constructor(data = {}) {
    this.subjectId = data.subject_id;
    this.name = data.name;

    this.universityId = data.university_id;
    this.universityName = data.university_name;

    this.departmentId = data.department_id;
    this.departmentName = data.department_name;

    this.subjectTypes = data.subject_types;

    this.credits = data.credits;

    this.processPercent = data.process_percent;
    this.midtermPercent = data.midterm_percent;
    this.finalPercent = data.final_percent;

    this.status = data.status;
  }
}

module.exports = SubjectDTO;